Source/ACE.Server/Entity/GeneratorProfile.cs

```diff
@@ -71,7 +72,7 @@ public int CurrentCreate
             get
             {
                 if (!GeneratedTreasureItem)
-                    return Spawned.Count + SpawnQueue.Count;
+                    return Spawned.Count + SpawnQueue.Count - RemoveQueue.Count(r => Spawned.ContainsKey(r.objectGuid));
```

Comment:
this is doing an o(n) lookup, which is no good. should be transformed into o(1)

The logic is also written in a way while it might technically be producing the correct outcome, semantically it is incorrect. It shouldn't be "counting a bool", it should be returning 1 if something exists, and 0 if something doesn't exist.